### PR TITLE
Rnd 24 cargo move

### PR DESCRIPTION
**Pull request by jakubbuskiewicz**
#### RELATED TICKETS

https://vortexa.atlassian.net/browse/RND-24

#### RELATED PRs

-

#### CHANGELOG

- added support for single cargo movement fetch with optional "params" which handles "unit"

#### TESTS

- I QA'd changes locally
- test cases were added

### CODE REVIEW

File: vortexasdk/operations.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def reference(self, id: ID) -> Dict:
             f"Looking up {self.__class__.__name__} reference data with id: {id}"
         )
 
-        data = default_client().get_reference(self._resource, id)
+        data = default_client().get_reference(self._ref_resource, id)
```
Comment (ministrudels) [inline vortexasdk/operations.py:45]: nit: can we revert back to `_resource`. Class is already named Reference

File: vortexasdk/endpoints/cargo_movements.py [hunk 2]
```diff
@@ -279,3 +283,19 @@ def search(
...
+            id: Cargo movement ID to lookup
+
+        # Returns
+        Cargo movement record matching the ID
+
+        # Further Documentation:
+        [VortexaAPI Cargo movement Reference](https://docs.vortexa.com/reference/GET/reference/)
+
+        """
+        return super().reference(id)
```
Comment (ministrudels) [inline vortexasdk/endpoints/cargo_movements.py:301]: `reference` refers to reference data, so I think this is a confusing API

**Review by migueluzcategui — APPROVED** [review]

File: vortexasdk/endpoints/cargo_movements.py [hunk 3]
```diff
@@ -279,3 +283,22 @@ def search(
...
+
+    def entity(self, id: ID, params: Dict = {}) -> Dict:
+        """
+        Perform a cargo movement lookup.
+
+        # Arguments
+            id: Cargo movement ID to lookup (long_id or short_id)
+
+            params: Supported search params:
+                unit: enter 'b' for barrels and 'b' for tonnes
```
Comment (ChristianBorresen) [inline vortexasdk/endpoints/cargo_movements.py:295]: I believe it should be b for barrels, t for tonnes, and cbm for cubic meters

File: vortexasdk/operations.py [hunk 4]
```diff
@@ -113,3 +113,80 @@ def search_with_client(
...
+
+    def entity_with_params(self, id: ID, params: Dict = {}) -> Dict:
+        """
+        Lookup for single entity using ID and search params.
+
+        # Arguments
+            id: Cargo movement ID to lookup (long_id or short_id)
+
+            params: Supported search params:
+                'unit': enter 'b' for barrels and 't' for tonnes
```
Comment (ChristianBorresen) [inline vortexasdk/operations.py:168]: also cbm

**Review by ChristianBorresen — CHANGES_REQUESTED** [review]

File: vortexasdk/operations.py [hunk 5]
```diff
@@ -113,3 +113,80 @@ def search_with_client(
...
+
+    def record(self, id: ID) -> Dict:
+        """
+        Lookup for single record using ID.
+
+        # Arguments
+            id: ID of the record we're looking up
+
+        # Returns
+        An record matching the ID
```
Comment (ChristianBorresen) [inline vortexasdk/operations.py:139]: "A record matching the ID"

File: vortexasdk/operations.py [hunk 6]
```diff
@@ -113,3 +113,80 @@ def search_with_client(
...
+        Lookup for single record using ID and search params.
+
+        # Arguments
+            id: Cargo movement ID to lookup (long_id or short_id)
+
+            params: Supported search params:
+                'unit': enter 'b' for barrels, 't' for tonnes and 'cbm' for cubic meters
+
+        # Returns
+        An record matching the ID
```
Comment (ChristianBorresen) [inline vortexasdk/operations.py:171]: Same as above

**Review by ChristianBorresen — APPROVED** [review]
Please fix the two small typos

**Pull request merged**